packages/salesforce/src/util.js

```diff
@@ -18,64 +16,33 @@ async function basicAuth(state) {
         error,
       });
     });
+  console.info(`Connecting to salesforce as ${username}.`);
 
-  console.info(`Connected to salesforce as ${username}.`);
-
-  return state;
-}
+  return conn;
+};
 
-function tokenAuth(state) {
-  const { instance_url, access_token, apiVersion } = state.configuration;
+export const tokenAuth = configuration => {
+  const { instance_url, access_token, apiVersion } = configuration;
 
-  connection = new Connection({
+  const conn = new Connection({
     instanceUrl: instance_url,
     accessToken: access_token,
     version: apiVersion,
   });
 
-  return state;
-}
-
-/**
- * Creates a connection to Salesforce using Basic Auth or OAuth.
- * @function createConnection
- * @private
- * @param {State} state - Runtime state.
- * @returns {State}
- */
-export function createConnection(state) {
-  if (connection) {
-    return state;
-  }
-
-  const { access_token } = state.configuration;
-
-  return access_token ? tokenAuth(state) : basicAuth(state);
-}
-
-/**
- * Removes state.connection from state.
- * @example
- * removeConnection(state)
- * @function
- * @private
- * @param {State} state
- * @returns {State}
- */
-export function removeConnection(state) {
-  connection = null;
-  return state;
-}
+  return conn;
+};
 
-export function getConnection() {
+export const checkConnection = async connection => {
+  // console.log(typeof (await connection));
```

Comment:
This log line needs removing

Reply:
This is outdated, it was remove long ago